### PR TITLE
Add span queries

### LINKED ISSUE #1316: Missing support for span queries
`span_*` queries are currently unimplemented in Elastica; this is evident looking into the QueryBuilder class which throw a `NotImplementedException` for every `span_` method.

Documentation on the span queries can be found here: https://www.elastic.co/guide/en/elasticsearch/reference/current/span-queries.html

### PR DESCRIPTION
**Pull request by alekitto**
Fixes #1316 

- [x] span_term
- [x] span_multi
- [x] span_first
- [x] span_near
- [x] span_or
- [x] span_containing
- [x] span_within
- [x] span_not
- [ ] field_masking_span

### CODE REVIEW

**Comment by ruflin** [comment]
https://github.com/ruflin/Elastica/pull/1320 is now merged. This probably needs rebase now.

**Comment by alekitto** [comment]
I've rebased and squashed my work.
The only thing left is the `field_masking_span`, but it's barely documented (there's only an example), so i think we could leave this out, as the implementation of the query class would be probably wrong.

File: lib/Elastica/Query/SpanContaining.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            $this->setBig($big);
+        }
+    }
+
+    /**
+     * @param AbstractSpanQuery $little
+     */
+    public function setLittle(AbstractSpanQuery $little)
+    {
+        $this->setParam('little', $little);
```
Comment (ruflin) [inline lib/Elastica/Query/SpanContaining.php]: We normal have a `return $this->setParam('little', $little);` in here so all setters return the object itself. This allows for chaining functions.

**Comment by alekitto** [comment]
Ok, fixed all the setters in SpanContaining, SpanNot, SpanWithin and SpanNear (missed in #1320)

**Comment by ruflin** [comment]
@alekitto In the diff Builder.php shows up and I'm not sure why?

**Comment by ruflin** [comment]
BTW: Could you also add an entry to the changelog?

**Comment by alekitto** [comment]
Builder.php is an empty file, I probably deleted it without noticing.
The Changelog already contains a line for Span* queries addition. Do I have to add another one for these three?

**Comment by ruflin** [comment]
Argh, it seems some things got mixed up in the Changelog :-( The reason an additional entry is needed because just yesterday happened a release. So this goes into the next release.

**Comment by ruflin** [comment]
I just pushed a change to the CHANGELOG to bring it up-to-date. Make sure to get the most recent one in before updating it.

**Comment by alekitto** [comment]
👍 Done.

**Pull request merged**